R/Quosure.R

```diff
@@ -0,0 +1,290 @@
+#' Reproducible class with environment and code.
+#'
+#' @slot code (`character`) to reproduce the environment
+#' @slot env (`environment`) environment which content was generated by the evaluation
+#' of the `code` slot.
+setClass(
+  "Quosure",
+  representation(code = "character", env = "environment"),
+  prototype(code = character(0), env = new.env(parent = parent.env(.GlobalEnv)))
```

Comment:
Please be aware that any library will not be attached to such environment we have to use sth like here https://github.com/insightsengineering/teal.data/blob/76c2e33f59057215569be210b54a10d28bc4f476/R/TealDatasetConnector_constructors.R#L827-L837

Example:

```r
ee <- new.env(parent = parent.env(globalenv()))
eval(quote({library(pacs); pac_description}), ee)

# VS

ee <- new.env(parent = parent.env(globalenv()))
ee$library <- function(...) {
  mc <- match.call()
  mc[[1]] <- quote(base::library)
  eval(mc, envir = globalenv())
  this_env <- parent.frame()
  if (!identical(this_env, globalenv())) {
    parent.env(this_env) <- parent.env(globalenv())
  }
}
eval(quote({library(cat2cat); cat2cat}), ee)
``

Reply:
I wonder if library calls should be treated differently (or even not allowed)...